app/components/WidgetDynamicContent.tsx

```diff
@@ -20,6 +20,7 @@ export default class WidgetDynamicContent extends Component {
         this.state = {
             loading: false
         };
+        this.attachedEvents = createRef<Record<string, any>>({});
```

Comment:
Minor:

Do we need a ref? I believe we could just have a regular property here :slightly_smiling_face:

Reply:
Another minor, if you decide to stick with `createRef` - from your comment in https://github.com/cloudify-cosmo/cloudify-stage/pull/1491#discussion_r669406841 it looks like `createRef` does not take any arguments :slightly_smiling_face: Thus, having `{}` is misleading

```suggestion
        this.attachedEvents = createRef<Record<string, any>>();
```